### PR TITLE
Implementation of POST /api/event/createEvent endpoint

### PR DESCRIPTION
**Pull request by xltvy**
This pull request is about the event creation operation, concerning a POST endpoint of our API called /event/createEvent. As discussed in #195 I will be implementing this POST endpoint and test my implementation using Postman.

After the implementation and testing stages are done, I will be preparing a documentation for the API regarding this POST endpoint.

### CODE REVIEW

**Comment by xltvy** [comment]
The documentation of this POST endpoint can be viewed [here](https://documenter.getpostman.com/view/21003322/UyxkkRM1). It is created via Postman.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencies": {
```
Comment (bahricanyesil) [inline package.json:2]: I guess this package.json file is created mistakenly. Can you check that out and delete it if it is created unintentionally.

File: practice-app/server/src/api/controllers/event/createEvent.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    const http_formatted_address = address.replace(/\s/g, '+');
+
+    var url = "https://maps.googleapis.com/maps/api/geocode/json?";
+
+    var params = {
+        address: http_formatted_address,
+        key: apiKeyGeo
+    };
+
+    Object.keys(params).forEach(function (url_key) { url += "&" + url_key + "=" + params[url_key]; });
```
Comment (bahricanyesil) [inline practice-app/server/src/api/controllers/event/createEvent.js:19]: We may make it like "{ url += url_key + "=" + params[url_key] + "&"; });" to start the query parameters without "&" at the beginning.

File: practice-app/server/src/api/controllers/event/createEvent.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    var lesson_flag = mongoose.Types.ObjectId.isValid(lesson_id);
+
+    if (!lesson_flag) {
+        throw new Error("Invalid lesson ID");
+    }
+
+    const host_lesson = await Lesson.findById(lesson_id).catch((err) => {
+        throw new Error("Lesson does not exist.");
```
Comment (bahricanyesil) [inline practice-app/server/src/api/controllers/event/createEvent.js:62]: This error message should be changed as something like "server error".

File: practice-app/server/src/api/controllers/event/createEventEndpoint.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import { createEvent } from "./createEvent.js";
+
+export default async (req, res) => {
+    try {
+        const {resultMessage, event} = await createEvent(req.body.title, req.body.date, req.body.location, req.body.host_id, req.body.lesson_id);
+        let status = 200;
+        console.log(resultMessage);
```
Comment (bahricanyesil) [inline practice-app/server/src/api/controllers/event/createEventEndpoint.js:7]: We can delete the console log lines.

File: practice-app/server/src/api/controllers/event/createEventEndpoint.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        return res.status(status).json({"resultMessage": resultMessage, "event": event});
+    }catch(err) {
+        let status = 500;
+        if(err.message == "\"title\" length must be at least 3 characters long")
```
Comment (bahricanyesil) [inline practice-app/server/src/api/controllers/event/createEventEndpoint.js:11]: Normally, returning the status codes in the places where we detected the errors is the preferred approach. However, this is OK for now for the practical purposes. So, you can just resolve this without taking any action.

**Review by bahricanyesil — CHANGES_REQUESTED** [review]

**Comment by xltvy** [comment]
Thank you for your review and suggestions @bahricanyesil , that json file is indeed created mistakenly alongside the console logs for the debugging. Now, I have made some changes according to your suggestions and now it can be reviewed one last time.

**Comment by bahricanyesil** [comment]
> Thank you for your review and suggestions @bahricanyesil , that json file is indeed created mistakenly alongside the console logs for the debugging. Now, I have made some changes according to your suggestions and now it can be reviewed one last time.

Thank you for the changes @xltvy. Did you check last time the query change we did? Did you try send query parameters and got the result as expected. If so, I will approve the pull request and merge it.

**Comment by xltvy** [comment]
> > Thank you for your review and suggestions @bahricanyesil , that json file is indeed created mistakenly alongside the console logs for the debugging. Now, I have made some changes according to your suggestions and now it can be reviewed one last time.
> 
> Thank you for the changes @xltvy. Did you check last time the query change we did? Did you try send query parameters and got the result as expected. If so, I will approve the pull request and merge it.

Yes, it runs smoothly.

**Review by bahricanyesil — APPROVED** [review]

**Pull request merged**